worlds/cccharles/__init__.py

```diff
@@ -0,0 +1,171 @@
+from .Items import CCCharlesItem, unique_item_dict, full_item_list, item_groups
+from .Locations import CCCharlesLocation, location_table
+from .Options import CCCharlesOptions
+from .Rules import set_rules
+from .Regions import CCCharlesRegion, create_regions
+from BaseClasses import Tutorial, ItemClassification
+from worlds.AutoWorld import World, WebWorld
+
+
+class CCCharlesWeb(WebWorld):
+    """
+    Choo-Choo Charles is a horror game.
+    A devil spider train from hell called Charles chases any person it finds on an island.
+    The goal is to gather scraps to upgrade a train to fight Charles and travel by train to find 3 eggs
+    to lead Charles to a brutal death and save the island.
+    """
+
+    theme = "stone"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setup Choo-Choo Charles for the Archipelago MultiWorld Randomizer.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Yaranorgoth"]
+    )
+
+    setup_fr = Tutorial(
+        "Guide d'Installation Multiworld",
+        "Un guide pour mettre en place Choo-Choo Charles pour le Randomiseur Multiworld Archipelago",
+        "Français",
+        "setup_fr.md",
+        "setup/fr",
+        ["Yaranorgoth"]
+    )
+
+    tutorials = [setup_en, setup_fr]
+
+    game_info_languages = ["en", "fr"]
+    rich_text_options_doc = True
+
+
+class CCCharlesWorld(World):
+    """ 
+    An independent 3D horror game, taking place on an island.
+    The main gameplay consists of traveling and fighting a monster on board a train.
+    Upgrading the train requires leaving the train to gather resources with the threat of encountering the monster.
+    """
+
+    game = "Choo-Choo Charles"
+
+    web = CCCharlesWeb()
+
+    item_name_to_id = unique_item_dict
+    location_name_to_id = location_table
+    item_name_groups = item_groups
+
+    # Options the player can set
+    options_dataclass = CCCharlesOptions
+    # Typing hints for all the options we defined
+    options: CCCharlesOptions
+
+    topology_present = True  # show path to required location checks in spoiler
+
+    def create_regions(self) -> None :
+        create_regions(self.multiworld, self.options, self.player)
+
+    def create_item(self, name: str) -> CCCharlesItem:
+        item_id = unique_item_dict[name]
+
+        match name:
+            case "Scraps" :
```

Comment:
```suggestion
        match name:
            case "Scraps":
```
Many (but not all) places in the code include a space before colons in blocks or dicts, while it's preferred for there not to be one. I suppose even if you prefer this way, the places where there isn't one should be changed for consistency.

Also when it comes to using a match here, it does maybe go against light guidance from the [style guide](https://github.com/duckboycool/Archipelago/blob/docs_typing/docs/style.md). Up to you if you think "they result in nicer code" here though really. I might prefer for having some sort of item data class like a lot of other worlds that contains the classification when listing them in `Items.py`.